### PR TITLE
Complete user registration, authentication

### PR DESCRIPTION
**Pull request by formalinfigs**
1) error handling in ctrl
2) naming of variables (be carefully)
3) status codes

### CODE REVIEW

File: data-access/repositories/booking.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        return booking;
+    }
+
+    async getAll(user) {
+        const bookings = this._mongooseModel.find({ user });
+
+        return { ...bookings };
+    }
+
+    getAllApprovedByUserIdAndFilters(user, isCompleted, city, country) {
```
Comment (artem-kushal) [inline data-access/repositories/booking.js]: combine with getAllUnapprovedByUserIdAndFilters

Comment (formalinfigs) [reply data-access/repositories/booking.js]: +

File: business-logic/services/booking-service.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        const bookings = await this._getBookingsInDateRange(roomId, dateFrom, dateTo);
+        const dates = this._getDatesOfRange(new Date(dateFrom), new Date(dateTo));
+
+        const room = await this._roomRepository.get(roomId);
+
+        return dates.filter((date) => this._isNotAvailableForBookingAtDate(date, bookings, room.count));
+    }
+
+    async isRoomAvailableInDateRange(roomId, dateFrom, dateTo) {
+        const bookings = await this._getBookingsInDateRange(roomId, dateFrom, dateTo);
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply business-logic/services/booking-service.js]: +

File: business-logic/models/booking.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    ) {
+        this.id = id;
+        this.userId = userId;
+        this.room = {
+            type: room.roomType,
+            capacity: room.capacity,
+            cost: room.cost
+        };
+        this.isApproved = isApproved;
+        this.hotel = {
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply business-logic/models/booking.js]: +

File: business-logic/models/booking.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        guests,
+        dateTo,
+        dateFrom,
+        arrivalTime,
+        departureTime,
+        facilities
+    ) {
+        this.id = id;
+        this.userId = userId;
+        this.room = {
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply business-logic/models/booking.js]: +

File: business-logic/services/booking-service.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+                    }
+                });
+
+            return documents.map((document) => this._model.parse(document));
+        } catch (err) {
+            // return err?
+        }
+    }
+
+    addDays(currentDate, days) {
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply business-logic/services/booking-service.js]: +

File: business-logic/services/booking-service.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                        path: 'hotel'
+                    }
+                })
+                .execPopulate()
+        );
+    }
+
+    // update selected services, arrival and departure time, totalPrice,
+    // isApproved flag, guests count
+    async update(data) {
```
[the remark this answers is not in the document]
Comment (formalinfigs) [reply business-logic/services/booking-service.js]: +

**Pull request merged**